tags.go

```diff
@@ -0,0 +1,89 @@
+package brokerTags
+
+import (
+	"time"
+)
+
+type TagManager struct {
+	broker          string
+	cfClientWrapper CFClientWrapper
+}
+
+func NewTagManager() (*TagManager, error) {
+	cfClientWrapper, err := NewCFClientWrapper()
+	if err != nil {
+		return nil, err
+	}
+	return &TagManager{
+		cfClientWrapper: cfClientWrapper,
+	}, nil
+}
+
+func (t *TagManager) GenerateTags(
+	action string,
+	serviceGUID string,
+	servicePlanGUID string,
+	organizationGUID string,
+	spaceGUID string,
+	instanceGUID string,
+) (map[string]string, error) {
+	tags := make(map[string]string)
+
+	tags["client"] = "Cloud Foundry"
+
+	tags["broker"] = t.broker
+
+	tags[action+" at"] = time.Now().Format(time.RFC822Z)
+
+	if serviceGUID != "" {
+		tags["Service GUID"] = serviceGUID
```

Comment:
I'm tempted to recommend factoring all the tag keys to consts, not because they're used in multiple places, but to make it extra clear to future developers that they shouldn't change. (If you use the `const ( ... )` syntax, you can apply one "don't change these!" comment to them all.) I'm thinking about how `net/http` uses (public int) consts for all the HTTP status codes. Thoughts?

Reply:
Ooh, another good reason: If you make them public, they're part of the public interface (like those status codes), and consumers of the packages can use them instead of string literals if they ever need to index into the map!